libraries/src/Document/HtmlDocument.php

```diff
@@ -794,6 +798,39 @@ protected function _renderTemplate()
 			$with[] = $this->getBuffer($args['type'], $args['name'], $args['attribs']);
 		}
 
+		$this->_template = str_replace('</body>', self::renderIcons() . '</body>', $this->_template);
+
 		return str_replace($replace, $with, $this->_template);
 	}
+
+
+	/**
+	 * Renders the document icons
+	 *
+	 * @return  string  The svgs markup for the icons
+	 *
+	 * @since   4.0.0
+	 */
+	public function renderIcons()
+	{
+		$files        = [];
+
+		// Generate the file and load the stylesheet link
+		foreach ($this->icons as $key => $icon)
+		{
+			$file = HTMLHelper::image('vendor/font-awesome/' . $icon . '.svg', '', null,true, 1);
+
+			if ($file) {
+				$matches = [];
+				preg_match('$(\/media.+)$', $file, $matches);
+				$files[] = @file_get_contents(JPATH_ROOT . $matches[1]);
```

Comment:
Indeed it does. Nice one.

If you want, you can do this as an oneliner `$files[] = file_get_contents(JPATH_ROOT . substr($file, strpos($file, '/media')));`

Added the same technique now in DPCalendar.